src/lib.rs

```diff
@@ -342,12 +400,22 @@ impl VadSession {
     /// them instead of just focusing on raw network output.
     pub fn current_speech_duration(&self) -> Duration {
         Duration::from_millis(
-            (self.current_speech_samples() / (self.config.sample_rate / 1000)) as u64,
+            (self.current_speech_samples() as f32 / self.config.sample_rate as f32 * 1000.0) as u64,
         )
     }
 
-    /// Get the current length of the VAD session.
+    /// Get the current duration of the VAD session, which includes both processed and unprocessed
+    /// samples.
     pub fn session_time(&self) -> Duration {
+        Duration::from_secs_f64(
+            (self.session_audio.len() as f64 + self.deleted_samples as f64)
```

Comment:
NIT: `(self.session_audio.len() + self.deleted_samples) as f64` is a b it less noisy and in some cases it will be more accurate (though likely never at the sort of sizes we expect)